libs/sdk-core/src/breez_services.rs

```diff
@@ -96,6 +96,34 @@ pub struct InvoicePaidDetails {
     pub bolt11: String,
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct SignMessageRequest {
+    /// The message to be signed
+    pub message: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SignMessageResponse {
+    /// The signature that covers the message of SignMessageRequest
+    pub signature: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CheckMessageRequest {
+    /// The message to be checked
+    pub message: String,
+    /// The public key
+    pub pubkey: String,
```

Comment:
Can a node verify signatures for multiple pubkeys? (multiple derivation paths?)

If not, isn't the pubkey here superfluous?

Reply:
It verifies that the given signature is a signature over the specified message, signed by the private key belonging to the passed pubkey.
So if another lightning node signs a message, you can verify it was that other lightning node signing the message. If it was not this pubkey that signed the message, the signature is invalid.
It's implemented this way in all lightning implementations, so this is cross compatible with cln, lnd, etc.

Reply:
Oh ok, so `check_message` can check any message signed by any node, not just the own node.

Reply:
Yeah exactly